app-creator/src/redux/types/attributes.ts

```diff
@@ -0,0 +1,283 @@
+  | 'textAlign'
+  | 'whiteSpace'
+  | 'shown';
+
+export interface AttributeMetaData {
+  [key: string]: {
+    value: string;
+    type: InputType;
+    items?: string[];
+  };
+}
+
+export interface DefaultAttributesType {
+  [key: string]: string;
+}
+
+export const sharedAttributes: AttributeMetaData = {
+  height: {
+    value: '',
+    type: InputType.number,
+  },
+  width: {
+    value: '',
+    type: InputType.number,
+  },
+  padding: {
+    value: '0',
+    type: InputType.number,
+  },
+  margin: {
+    value: '8',
+    type: InputType.number,
+  },
+  color: {
+    value: 'black',
+    type: InputType.color,
+  },
+  backgroundColor: {
+    value: 'none',
+    type: InputType.color,
+  },
+  borderWidth: {
+    value: '0',
+    type: InputType.number,
+  },
+  borderType: {
+    value: 'solid',
+    type: InputType.select,
+    items: ['solid', 'dashed'],
+  },
+  borderColor: {
+    value: 'black',
+    type: InputType.color,
+  },
+  fontSize: {
+    value: '12',
+    type: InputType.number,
+  },
+  fontWeight: {
+    value: '300',
+    type: InputType.select,
+    items: ['100', '200', '300', '400', '500', '600', '700', '800', '900'],
+  },
+  fontFamily: {
+    value: 'Roboto',
+    type: InputType.select,
+    items: [
+      'Arial',
+      'Arial Black',
+      'Bookman',
+      'Candara',
+      'Comic Sans MS',
+      'Courier',
+      'Courier New',
+      'Garamond',
+      'Georgia',
+      'Impact',
+      'Open Sans',
+      'Palatino',
+      'Roboto',
+      'Times',
+      'Times New Roman',
+      'Verdana',
+    ],
+  },
+  textAlign: {
+    value: 'left',
+    type: InputType.select,
+    items: ['left', 'center', 'right', 'justify'],
+  },
+  whiteSpace: {
+    value: 'normal',
+    type: InputType.select,
+    items: ['normal', 'pre'],
+  },
+  shown: {
+    value: 'true',
+    type: InputType.select,
+    items: ['true', 'false'],
+  },
+};
+
+export const DEFAULT_SHARED_ATTRIBUTES: DefaultAttributesType = Object.keys(
+  sharedAttributes
+).reduce((attributes, key) => {
+  if (sharedAttributes[key].type === InputType.number) {
+    return { ...attributes, [key]: sharedAttributes[key].value + 'px' };
+  }
+  return { ...attributes, [key]: sharedAttributes[key].value };
+}, {});
+
+export const textboxAttributes: AttributeMetaData = {
+  value: {
+    value: '',
+    type: InputType.text,
+  },
+  placeholder: {
+    value: 'Enter text',
+    type: InputType.text,
+  },
+};
+
+export const DEFAULT_TEXTBOX_ATTRIBUTES: DefaultAttributesType = Object.keys(
+  textboxAttributes
+).reduce(
+  (attributes, key) => ({ ...attributes, [key]: textboxAttributes[key].value }),
+  {}
+);
+
+export const labelAttributes: AttributeMetaData = {
+  value: {
+    value:
+      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
```

Comment:
"Google Earth Engine combines a multi-petabyte catalog of satellite imagery and geospatial datasets with planetary-scale analysis capabilities and makes it available for scientists, researchers, and developers to detect changes, map trends, and quantify differences on the Earth's surface."
From: http://earthengine.google.com/ :)